### PR TITLE
Fixed joint war.

### PR DESCRIPTION
**Pull request by pspjuth**
This makes it possible to get joint war and research agreements in vanilla version.
This adds back the initial dialog from unmodded trade screen.
It seems the deal manager refuses incomplete deals and thus the upfront dialog is needed.
This solution should work for R&F as well with Third Party and Casus Belli choices.

### CODE REVIEW

**Comment by pspjuth** [comment]
This fixes the last problem in vanilla I'm aware of.

File: Integrations/IDS/diplomacydealview.xml [hunk 1]
```diff
@@ -229,6 +238,59 @@
...
+                          <Grid Style="DropShadow3" Size="parent,50"/>
+                          <Grid Anchor="C,C" Texture="Controls_WindowLight" SliceTextureSize="34,34" SliceCorner="17,17" Size="parent-30,34">
+                            <Grid Anchor="C,C" Texture="Controls_TextArea" SliceTextureSize="22,22" SliceCorner="11,20" Size="parent-20,24">
+                              <EditBox ID="ValueAmountEditBox" Style="FontNormalBold16" FontStyle="Stroke" NumberInput="1" EditMode="1" Size="parent-5,parent" Anchor="C,C" MaxLength="5" ConsumeMouse="1" HighlightColor="25,120,154,200"/>
+                              <Button ID="ValueAmountEditLeftButton" Style="ArrowButtonLeft" Anchor="L,C" AnchorSide="O,I"/>
+                              <Button ID="ValueAmountEditRightButton" Style="ArrowButtonRight" Anchor="R,C" AnchorSide="O,I"/>
+                            </Grid>
+                          </Grid>
+                        </Container>
+                      </Stack>
```
Comment (ihendriks) [inline Integrations/IDS/diplomacydealview.xml]: from L258 until here the code is indented too much (2 spaces)

**Comment by pspjuth** [comment]
I think since the Improved Deal Screen had a very different design, this popup was removed.
And that worked up to the last patch where it seems something changed in the deal manager.

**Comment by ihendriks** [comment]
Looks  good to me now.

And yeah, that does make the most sense. Just seemed like a lot of lua was missing as well, which seems weird to me, as it's usually commented out with a note instead of just removed. But maybe it's because it's an integration, and not native CQUI.

**Comment by Azurency** [comment]
Thank you for your help 👍. However the popup was removed, and I'd like to keep it that way as it fits the orignal spirit of the improved deal screen better.
You guested it right, the code was removed by the original creator of this mod.
I just pushed a fix for the base game and I'll test it for R&F when everything is merged together on the expansion1 branch.

**Pull request closed without merging**